### PR TITLE
Add Yarn workspaces

### PR DESCRIPTION
**Pull request by it-harrison**
## Description
This PR adds Yarn workspaces to `vets-website`. See https://depo-platform-documentation.scrollhelp.site/developer-docs/yarn-workspaces 

## Original issue(s)
department-of-veterans-affairs/va.gov-team#39205

## Testing done
Local testing passed. CI testing passed. 

## Screenshots
N/A

## Acceptance criteria
- [x] isolated dependencies in an application with workspaces are installed to a `node_modules` directory inside the application's workspace
- [x] `platform` exports can be imported via the yarn import syntax
- [x] scripts inside an application workspace can be run from the command line
- [x] applications that do not use workspaces are not affected by the addition of workspaces to the repo

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by Mottie — APPROVED** [review]

**Review by micahchiang — APPROVED** [review]

**Review by timwright12 — CHANGES_REQUESTED** [review]
Let's get the PR filled out as well as a test plan for these changes since they're so massive. We should also think about breaking this up into a series of smaller PRs

**Comment by AGarcia-Clarity** [comment]
Just in case anyone else needs help knowing how to test this PR. I followed the [Workspace Maintenance Testing instructions](https://depo-platform-documentation.scrollhelp.site/developer-docs/guide-for-maintainers-overview), additionally tested [the new workspace scripts](https://depo-platform-documentation.scrollhelp.site/developer-docs/yarn-workspace-utility-scripts#YarnWorkspaceUtilityScripts-list-imports:appsandlist-imports:plat), and verified no obvious errors.


## Overall testing
1. Delete all pertinent `node_modules` folders

2. Run `yarn install`

3. Check `./node_modules/@department-of-veterans-affairs/applications-<your-application-name>`  exists.
    1. Example that has a workspace: `./node_modules/@department-of-veterans-affairs/platform-user` (has package.json in directory)
    2. Example that DOES not have a workspace: `./node_modules/@department-of-veterans-affairs/applications-login` (does not have package.json in directory)
    
4. Check your application’s root `node_modules` to make sure non-hoisted dependencies are present and correct

5. Run `yarn watch` to test for compilation and runtime errors; the yarn workspaces installation should NOT affect the import/export running of current code (only provide a new option)

6. Run unit tests

7. Confirm imports using the format @department-of-veterans-affairs/... function properly

## Scripts testing
1. Run `yarn list-imports:apps --app-folder <your team's application folder>`
    1. Example: `yarn list-imports:apps --app-folder login`
2. Run `yarn list-imports:plat --app-folder <your team's platform folder>`
    1. Example: `yarn list-imports:plat --app-folder user/authentication`
3. Testing workspaces are working by run `demo` command
    1. Example: `yarn workspace @department-of-veterans-affairs/platform-utilities run demo`

**Review by asg5704 — APPROVED** [review]
Looks good.  Next time can you include instructions on how to test this PR.

File: src/platform/user/profile/vap-svc/actions/transactions.js [hunk 1]
```diff
@@ -4,12 +4,15 @@ import {
   ADDRESS_POU,
   FIELD_NAMES,
   TRANSACTION_STATUS,
-} from '@@vap-svc/constants';
-import { showAddressValidationModal, inferAddressType } from '@@vap-svc/util';
-import { hasBadAddress } from '@@profile/selectors';
```
Comment (adamwhitlock1) [inline src/platform/user/profile/vap-svc/actions/transactions.js:9]: Are we not going to be able to use the `@@profile` and `@@vap-svc` anymore?

Comment (it-harrison) [reply src/platform/user/profile/vap-svc/actions/transactions.js:9]: That's something that may have escaped the documentation (but which we will fix). Inside a workspace import paths with those aliases are not resolved by eslint, so we removed them.

File: src/platform/utilities/demo-script/local-demo.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable no-console */
+const commandLineArgs = require('command-line-args');
+
+const args = commandLineArgs([{ name: 'from', type: Number }]);
```
Comment (alexandec) [inline src/platform/utilities/demo-script/local-demo.js:4]: Is this script supposed to be here or is it just for testing?

File: src/platform/utilities/package.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        "date-fns": "*",
+        "@department-of-veterans-affairs/component-library": "*",
+        "@sentry/browser": "*", 
+        "chai": "*", 
+        "sinon": "*" 
+    },
+    "license": "MIT",
+    "private": true
+   }
+
```
Comment (alexandec) [inline src/platform/utilities/package.json:50]: Noticed a bunch of files missing the final newline, could correct in new PRs

**Review by adamwhitlock1 — APPROVED** [review]

File: src/platform/utilities/demo-script/local-demo.js (hunk 2 above)
Comment (rmessina1010) [reply src/platform/utilities/demo-script/local-demo.js:4]: This is just for testing/demoing.

**Review by k80bowman — APPROVED** [review]

**Pull request closed without merging**